src/client.c

```diff
@@ -1705,22 +1727,38 @@ static gboolean add_recipe_host(const gchar *value, AppData *app_data,
     gchar *connect_uri = NULL;
     gchar *id = NULL;
     gboolean result = TRUE;
+    gchar **ssh_destination;
 
     args = g_strsplit(value, "=", 2);
     if (g_strv_length(args) != 2) {
         id = g_strdup_printf("%u", (*recipe_id)++);
-        connect_uri = g_strdup (args[0]);
+        connect_uri = args[0];
     } else {
-        id = g_strdup(args[0]);
-        connect_uri = g_strdup (args[1]);
+        id = args[0];
+        connect_uri = args[1];
     }
 
-    RecipeData *recipe_data = new_recipe_data(app_data, id);
-    recipe_data->connect_uri = connect_uri;
-    recipe_data->rhost = parse_host(recipe_data->connect_uri);
+    ssh_destination = parse_host (connect_uri);
+
+    if (ssh_destination != NULL) {
+        RecipeData *recipe_data;
+
+        recipe_data = new_recipe_data (app_data, id);
+        recipe_data->connect_uri = g_strjoinv (NULL, ssh_destination);
+        recipe_data->rhost = ssh_destination[1];
+
+        g_free (ssh_destination[0]);
+        g_free (ssh_destination);
+
+        result = TRUE;
+    } else {
+        result = FALSE;
+    }
 
     g_free (id);
-    g_strfreev (args);
+    g_free (connect_uri);
```

Comment:
If you are no longer doing g_strdup, do you still need to free connect_uri?

Reply:
We are freeing the strings in `args` here.

Reply:
I believe you missed my question.  Not sure how connect_uri variable relates to args in your response.